### PR TITLE
feat(Button): Loading button with spinner and aria-disabled

### PR DESCRIPTION
**Pull request by allinox**
These are suggested changes to the button component related to issue #204 

It includes:
Using aria-disabled instead of disabled for button to make it more accessible.
Added loading button example to storybook
Made title optional on Spinner (it is obsolete in cases like these)

### CODE REVIEW

File: packages/react/src/components/Button/Button.stories.mdx [hunk 1]
```diff
@@ -194,6 +208,44 @@ On through of which no the your phase more know anyone tried he when line me wha
     </Story>
 </Canvas>
 
+### Knapp som laster
+<Canvas>
```
Comment (mimarz) [inline packages/react/src/components/Button/Button.stories.mdx:212]: Suggest adding a line or two about using `aria-disabled` instead of `disabled` for loading buttons, but don't worry if you can't come up with the wording, We can update it later if needed.

File: packages/react/src/components/Button/Button.tsx [hunk 2]
```diff
@@ -70,6 +72,8 @@ const Button = (
       className,
     )}
     type={type}
+    aria-disabled={disabled}
+    onClick={disabled ? undefined : onClick}
```
Comment (mimarz) [inline packages/react/src/components/Button/Button.tsx:76]: I suggest removing this and just sticking to updating the css selectors. This might be confusing for anyone passing in `disabled` and having something else applied instead. Suggested behaviour here should be documented instead (such as developer has to make sure to disable `onClick` if they are using `aria-disabled`). We want our `Button` to behave as native as possible.

File: packages/react/src/components/Spinner/Spinner.tsx [hunk 3]
```diff
@@ -41,7 +41,7 @@ const variantMap: Record<Variant, { foreground: string; background: string }> =
 
 export type SpinnerProps = {
   /** Spinner title  */
-  title: string;
+  title?: string;
```
Comment (mimarz) [inline packages/react/src/components/Spinner/Spinner.tsx:44]: I don't see a problem with this, but I would suggest its done a separate PR for traceability. We automate our changelogs where each PR is a separate entry so its also clearer for users of changes made.

**Review by mimarz — CHANGES_REQUESTED** [review]
Thank you so much for taking the time to do these updates! 👏 

We do however still need the default `disabled` to work as before, so the css selectors should be added alongside disabled instead of replacing them :)

We use [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) formatting for our PRs for better traceability and automatic changelogs so suggest updating the starting the title with something like "feat(Button)", since we are adding the feature of styling like a disabled button on `aria-disabled`

File: packages/react/src/components/Button/Button.tsx (hunk 2 above)
Comment (allinox) [reply packages/react/src/components/Button/Button.tsx:76]: Ahh, ok, I thought we were aiming to create something where the users of the design system could use it straight forward and know that it will simply be accessible out of the box, but I agree this could be confusing for those that do want to use the disabled functionality as-is 😅
I'll fix it when I get back from DevOps days later this week 😊

**Comment by mimarz** [comment]
I've made a few changes to linting and other tools, so I recommend you rebase your branch with the latest changes before you continue! 😄

**Comment by allinox** [comment]
Will do - thanks for the heads up! 😊

**Comment by allinox** [comment]
Hmm... This rebasing exactly go as planned... 😅
Might kill this PR and make one that's easier to review.
(But that'll have to wait for tomorrow)

**Comment by mimarz** [comment]
> Hmm... This rebasing didn't exactly go as planned... 😅 Might kill this PR and make one that's easier to review. (But that'll have to wait for tomorrow)

Hehe, I know the feeling! 

In short when rebasing a branch you can do the following:

- Navigate to the branch you want to rebase
- `git fetch` ⬇️ 
- `git rebase origin/main` (or just `git rebase main` if you have pulled the latest changes to your local `main` branch)
- Resolve any conflicts that may arise 👷 
- Add the fixed file with `git add .` (or just + in vscode source control if you are using that)
- `git rebase --continue`
  - or `git rebase --abort` if things gets borked
- After the rebase is done, git will now tell you that your branch is out of sync. 
- Ignore that and run `git push -f` (or `git push --force-with-lease` if you want to be safe) 💪 
- Enjoy updated branch with latest changes, celebrate with coffee and chocolate ☕ 🍫 

More details can also be found in [this guide](https://www.w3docs.com/snippets/git/how-to-rebase-git-branch.html) :)

**Comment by allinox** [comment]
Ahh, yeah, I didn't do the forcing - I thought git would realize there'd been a rebase by itself 😅
I'll save your nifty guide for next time! For now though, I think I'll just kill this PR 💀
